File: server/src/main/scala/kpn/api/common/node/NodeDetailsPage.scala
package kpn.api.common.node

import kpn.api.common.NodeInfo
import kpn.api.common.common.Reference

case class NodeDetailsPage(
  nodeInfo: NodeInfo,
  mixedNetworkScopes: Boolean,
  routeReferences: Seq[Reference],
  networkReferences: Seq[Reference],
  integrity: Option[NodeIntegrity],
  changeCount: Long
)

File: backend/api/src/main/java/kpn/api/common/node/NodeDetailsPage.java
package kpn.api.common.node;

import kpn.api.common.NodeInfo;
import kpn.api.common.common.Reference;
import kpn.api.common.node.NodeIntegrity;

import java.util.Optional;
import com.google.common.collect.ImmutableList;

public record NodeDetailsPage(
  NodeInfo nodeInfo,
  Boolean mixedRouteScopes,
  ImmutableList<Reference> routeReferences,
  ImmutableList<Reference> networkReferences,
  Optional<NodeIntegrity> integrity,
  Long changeCount
) {}
